src/nucleus/tt.ml

```diff
@@ -641,8 +641,9 @@ and print_share ~penv lshare ppf = match lshare with
 and print_shares ~penv lshares ppf = match lshares with
   | [] -> ()
   | [lshare] -> print_share ~penv lshare ppf
-  | ((_,Inl x) as lshare) :: lshares ->
-    Format.fprintf ppf "%t and %t" (print_share ~penv lshare) (print_shares ~penv:(add_forbidden x penv) lshares)
+  | (l,Inl x) :: lshares ->
+    let x = Name.refresh penv.forbidden x in
+    Format.fprintf ppf "%t and %t" (print_share ~penv (l,Inl x)) (print_shares ~penv:(add_forbidden x penv) lshares)
```

Comment:
In general, whenever you print a space you should give the formatter a chance to break the line by printing `@ ` (`@` followed by a space), so `"%t@ and@ %t"` here, and also in `print_share` there is an opportunity to put in `@ ` around `=`.